### PR TITLE
NXP: LPCXpresso55S16: Fix GPIO IRQs, enable I2C and SPI

### PR DESCRIPTION
**Pull request by henrikbrixandersen**
This patchset enabled the following functionality on the NXP LPCXpresso55S16 development board:
- Fixes GPIO interrupts (bug fix)
- Enables I2C and FXOS8700 support
- Enabled high-speed SPI support

### CODE REVIEW

File: boards/arm/lpcxpresso55s16/Kconfig.defconfig [hunk 1]
```diff
@@ -28,4 +28,16 @@ config GPIO_MCUX_LPC_PORT1
 
 endif # GPIO_MCUX_LPC
 
+config I2C_4
```
Comment (galak) [inline boards/arm/lpcxpresso55s16/Kconfig.defconfig:31]: Are these needed the (I2C_4 and SPI_8)?

**Review by galak — COMMENTED** [review]
LGTM, just curious if we really need the SPI_8 and I2C_4 in the board Kconfig

File: boards/arm/lpcxpresso55s16/Kconfig.defconfig (hunk 1 above)
Comment (henrikbrixandersen) [reply boards/arm/lpcxpresso55s16/Kconfig.defconfig:31]: For now they are (due to https://github.com/zephyrproject-rtos/zephyr/blob/28e2bfc68e728410ee82f81cb6db42b3e7438dc7/soc/arm/nxp_lpc/lpc55xxx/soc.c#L64 and https://github.com/zephyrproject-rtos/zephyr/blob/28e2bfc68e728410ee82f81cb6db42b3e7438dc7/soc/arm/nxp_lpc/lpc55xxx/soc.c#L72).

File: boards/arm/lpcxpresso55s16/pinmux.c [hunk 2]
```diff
@@ -118,6 +118,35 @@ static int lpcxpresso_55s16_pinmux_init(struct device *dev)
 	pinmux_pin_set(port0, 30, port0_pin30_config);
 #endif
 
+#if DT_HAS_NODE(DT_NODELABEL(flexcomm4)) && \
+    DT_NODE_HAS_COMPAT(DT_NODELABEL(flexcomm4), nxp_lpc_i2c)
```
Comment (MaureenHelm) [inline boards/arm/lpcxpresso55s16/pinmux.c:122]: Please fix the leading spaces flagged by checkpatch

**Review by MaureenHelm — CHANGES_REQUESTED** [review]
Just one nit

File: boards/arm/lpcxpresso55s16/pinmux.c (hunk 2 above)
Comment (henrikbrixandersen) [reply boards/arm/lpcxpresso55s16/pinmux.c:122]: I can do that, but this is the coding style used for these checks in other files within the tree:
```
$ git grep '^    DT_NODE_HAS_COMPAT'|wc -l
11
```

File: boards/arm/lpcxpresso55s16/Kconfig.defconfig (hunk 1 above)
Comment (galak) [reply boards/arm/lpcxpresso55s16/Kconfig.defconfig:31]: Gotcha, will send PR to change that

**Comment by galak** [comment]
I merged #24980 so you can rebase this to remove setting I2C_4 & SPI_8.

**Comment by henrikbrixandersen** [comment]
> I merged #24980 so you can rebase this to remove setting I2C_4 & SPI_8.

@galak Thank you. I have rebased and removed the settings.

**Review by galak — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**